builtin/docker/platform.go

```diff
@@ -95,6 +104,136 @@ func (p *Platform) resourceManager(log hclog.Logger) *resource.Manager {
+	}
+
+	// Create our status report
+	var result sdk.StatusReport
+	result.External = true
+
+	// NOTE(briancain): The docker platform currently only deploys a single
+	// container, so for now the status report makes the same assumption.
+
+	log.Debug("querying docker for container health")
+
+	resources := []*sdk.StatusReport_Resource{{
+		Name: containerInfo.Name,
+	}}
+
+	if containerInfo.State.Health != nil {
+		// Built-in Docker health reporting
+		// NOTE: this only works if the container has configured health checks
+
+		switch containerInfo.State.Health.Status {
+		case "Healthy":
+			resources[0].Health = sdk.StatusReport_READY
+			resources[0].HealthMessage = "container is running"
+		case "Unhealthy":
+			resources[0].Health = sdk.StatusReport_DOWN
+			resources[0].HealthMessage = "container is down"
+		case "Starting":
+			resources[0].Health = sdk.StatusReport_ALIVE
+			resources[0].HealthMessage = "container is starting"
+		default:
+			resources[0].Health = sdk.StatusReport_UNKNOWN
+			resources[0].HealthMessage = "unknown status reported by docker for container"
+		}
+	} else {
+		// Waypoint container inspection
+
+		if containerInfo.State.Running && containerInfo.State.ExitCode == 0 {
+			resources[0].Health = sdk.StatusReport_READY
+			resources[0].HealthMessage = "container is running"
+		} else if containerInfo.State.Restarting || containerInfo.State.Status == "created" {
+			resources[0].Health = sdk.StatusReport_ALIVE
+			resources[0].HealthMessage = "container is still starting"
+		} else if containerInfo.State.Dead || containerInfo.State.OOMKilled || containerInfo.State.ExitCode != 0 {
+			resources[0].Health = sdk.StatusReport_DOWN
+			resources[0].HealthMessage = "container is down"
+		} else {
+			resources[0].Health = sdk.StatusReport_UNKNOWN
+			resources[0].HealthMessage = "unknown status for container"
+		}
+	}
+
+	result.Resources = resources
+
+	// Determine overall deployment health based on its resource health
+	var ready, alive, down, unknown int
+	for _, r := range result.Resources {
+		switch r.Health {
+		case sdk.StatusReport_DOWN:
+			down++
+		case sdk.StatusReport_UNKNOWN:
+			unknown++
+		case sdk.StatusReport_READY:
+			ready++
+		case sdk.StatusReport_ALIVE:
+			alive++
+		}
+	}
+
+	if ready == len(result.Resources) {
+		result.Health = sdk.StatusReport_READY
+	} else if down == len(result.Resources) {
+		result.Health = sdk.StatusReport_DOWN
+	} else if unknown == len(result.Resources) {
+		result.Health = sdk.StatusReport_UNKNOWN
+	} else if alive == len(result.Resources) {
+		result.Health = sdk.StatusReport_ALIVE
+	} else {
+		result.Health = sdk.StatusReport_PARTIAL
+	}
+
+	result.TimeGenerated = ptypes.TimestampNow()
+	log.Debug("status report complete")
+
+	// TODO(briancain): remove me GH #1469
+	time.Sleep(500 * time.Millisecond)
+
+	// update output based on main health state
+	s.Update("Finished building report for Docker platform")
+	s.Done()
+
+	st := ui.Status()
+	defer st.Close()
```

Comment:
Per the discussion last week, don't use Status, use the same stepgroup you have here and set the step's status.

Reply:
Ah yeah, switching back from Status to StepGroups, I remember now why I tried using the `ui.Status` directly. Every time I do something like this:

```golang
s.Update("some text to be printed!!")
s.Status(terminal.StatusOk)
```

The text shows up as just `ok` and not `"some text to be printed!!"`. Additionally if I do something like:

```golang
s.Update("some text to be printed!!")
s.Status(terminal.StatusWarn)
```

It just prints `warn` in a green status ok format.

Reply:
Quick note from our video sync: This code should be working as mentioned above, but the behavior described sounds like a bug. The right move is to still move away from Status, but we should address the output bug too.